File: G/cpp/code.cpp
#include <vector>
#include <iostream>

using namespace std;

struct Vertex {
    int w;
    int p;

    Vertex(int w, int p) {
        w = w;
        p = p;
    }
};


int getNumberOfUpgoingPaths(vector<Vertex> tree, int x) {
    // your code goes here
    return 0;
}

vector<Vertex> readTree(int n) {
    vector<Vertex> tree;
    for (int i = 0; i < n; i++) {
        int parent, weight;
        cin >> parent >> weight;
        tree.push_back(Vertex(weight, parent));
    }
    return tree;
}

int main() {
    int n;
    cin >> n;
    int x;
    cin >> x;
    vector<Vertex> tree = readTree(n);
    cout << getNumberOfUpgoingPaths(tree, x);

}

File: G/js/code.js
class Vertex {
    constructor(weight, parent) {
        this.weight = weight;
        this.parent = parent;
    }
}

function getNumberOfUpgoingPaths(tree, x) {
    // your code goes here
    return 0;
}

const _readline = require('readline');

const _reader = _readline.createInterface({
    input: process.stdin
});

const _inputLines = [];
let _curLine = 0;

_reader.on('line', line => {
    _inputLines.push(line);
});

process.stdin.on('end', solve);


function solve() {
    const firsLine = readArray();
    const n = firsLine[0];
    const x = firsLine[1];
    const tree = readTree(n);
    const ans = getNumberOfUpgoingPaths(tree, x);
    console.log(ans);    
}

function readInt() {
    const n = Number(_inputLines[_curLine]);
    _curLine++;
    return n;
}

function readArray() {
    var arr = _inputLines[_curLine].trim(" ").split(" ").map(num => Number(num));
    _curLine++;
    return arr;
}

function readTree(n) {
    let tree = [];
    for (let i = 0; i < n; i++) {
        let vertex = readArray();
        tree.push(new Vertex(vertex[1], vertex[0]));
    }
    return tree;
}